proc/proc_darwin.c

```diff
@@ -68,6 +70,29 @@ acquire_mach_task(int tid,
 	return mach_port_move_member(self, *notification_port, *port_set);
 }
 
+kern_return_t
+reacquire_mach_task(task_t task, mach_port_t *exception_port, mach_port_t *notification_port) {
+	kern_return_t kret;
+	mach_port_t prev_not;
+	mach_port_t self = mach_task_self();
+	
+	kret = task_set_exception_ports(task, EXC_MASK_BREAKPOINT|EXC_MASK_SOFTWARE, *exception_port,
+			EXCEPTION_DEFAULT, THREAD_STATE_NONE);
+	if (kret != KERN_SUCCESS) {
+		printf("task_set_exception_ports: %d\n", kret);
```

Comment:
nit: If you plan on keeping this logging (probably not a bad idea), I'd say something more along the lines of:

("error during reaquire_mach_task: could not set exception ports: %s", mach_error_string(kret))